planetiler-core/src/main/java/com/onthegomap/planetiler/archive/ReadableTileArchive.java

```diff
@@ -0,0 +1,44 @@
+package com.onthegomap.planetiler.archive;
+
+import com.onthegomap.planetiler.geo.TileCoord;
+import com.onthegomap.planetiler.util.CloseableIterator;
+import java.io.Closeable;
+
+/**
+ * Read API for on-disk representation of a tileset in a portable format. Example: MBTiles, a sqlite-based archive
+ * format.
+ * <p>
+ * See {@link WriteableTileArchive} for the write API.
+ */
+public interface ReadableTileArchive extends Closeable {
+
+  /** Returns the raw tile data associated with the tile at {@code coord}. */
+  default byte[] getTile(TileCoord coord) {
+    return getTile(coord.x(), coord.y(), coord.z());
+  }
+
+  /** Returns the raw tile data associated with the tile at coordinate {@code x, y, z}. */
+  byte[] getTile(int x, int y, int z);
```

Comment:
How should this handle the case of the tile not existing? should it be an option return type? It doesn't seem exceptional.

Reply:
Ah sorry the javadoc should indicate this, but it should return `null` if not found.

Reply:
Opened #486 to fix the javadoc.